engine-precompiles/src/modexp.rs

```diff
@@ -142,13 +145,16 @@ impl ModExp<Berlin> {
 impl Precompile for ModExp<Berlin> {
     fn required_gas(input: &[u8]) -> Result<EthGas, ExitError> {
         let (base_len, exp_len, mod_len) = parse_lengths(input);
+        if base_len == 0 && mod_len == 0 {
+            Ok(EthGas::new(200u64)) // min gas 200
```

Comment:
It's better to use constants in such cases rather than magic numbers. 
```rust
const MIN_GAS: u64 = 200;
```

Reply:
And to do it in every `impl` block for `ModExp<_>` struct. E.g.
```rust
impl ModExp<Byzantium> {
    const MIN_GAS: u64 = 0;

    // output of this function is bounded by 2^128
    fn mul_complexity(x: u64) -> U256 {
        if x <= 64 {
            U256::from(x * x)
        } else if x <= 1_024 {
            U256::from(x * x / 4 + 96 * x - 3_072)
        } else {
            // up-cast to avoid overflow
            let x = U256::from(x);
            let x_sq = x * x; // x < 2^64 => x*x < 2^128 < 2^256 (no overflow)
            x_sq / U256::from(16) + U256::from(480) * x - U256::from(199_680)
        }
    }
}

impl ModExp<Berlin> {
    const MIN_GAS: u64 = 200;

    // output bounded by 2^122
    fn mul_complexity(base_len: u64, mod_len: u64) -> U256 {
        let max_len = core::cmp::max(mod_len, base_len);
        let words = U256::from(Integer::div_ceil(&max_len, &8));
        words * words
    }
}
```

Reply:
But it will not be found in the scope of the `Precompile` impl

Reply:
It will be
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5dce524e9c06b126e455296afbfba024

Reply:
```rust
Ok(EthGas::new(Self::MIN_GAS))
```

Reply:
The constant could be `EthGas` since the `new` method of `EthGas` is const.
```rust
const MIN_GAS: EthGas = EthGas::new(200);
```